File: sueldo empleado java/arl.java
public class arl {
    private double baseSalary;
    private double percentage;

    public arl() {
        this.baseSalary = 0.0;
        this.percentage = 0.005; // Porcentaje predeterminado para ARL (0.5%)
    }

    /**
     * Establece el salario base.
     * 
     * @param baseSalary El salario base del empleado.
     * @throws IllegalArgumentException Si el salario base es negativo.
     */
    public void setBaseSalary(double baseSalary) {
        if (baseSalary < 0) {
            throw new IllegalArgumentException("El salario base no puede ser negativo.");
        }
        this.baseSalary = baseSalary;
    }

    /**
     * Obtiene el salario base.
     * 
     * @return El salario base.
     */
    public double getBaseSalary() {
        return baseSalary;
    }

    /**
     * Establece el porcentaje de deducción.
     * 
     * @param percentage El porcentaje de deducción (entre 0 y 1).
     * @throws IllegalArgumentException Si el porcentaje no está en el rango válido.
     */
    public void setPercentage(double percentage) {
        if (percentage < 0 || percentage > 1) {
            throw new IllegalArgumentException("El porcentaje debe estar entre 0 y 1.");
        }
        this.percentage = percentage;
    }

    /**
     * Obtiene el porcentaje de deducción.
     * 
     * @return El porcentaje de deducción.
     */
    public double getPercentage() {
        return percentage;
    }

    /**
     * Calcula las deducciones de ARL basadas en el salario base y el porcentaje.
     * 
     * @return El monto de las deducciones.
     */
    public double calculatedDeductions() {
        return baseSalary * percentage;
    }
}

File: sueldo empleado - copia/arl.php
<?php

require_once 'Deductions.php';

class Arl extends Deductions {
    private int $risk = 0;
    private array $riskPercentages = [
        1 => 0.00522, // Riesgo 1: 0.522%
        2 => 0.01044, // Riesgo 2: 1.044%
        3 => 0.02436, // Riesgo 3: 2.436%
        4 => 0.04350, // Riesgo 4: 4.350%
        5 => 0.06960  // Riesgo 5: 6.960%
    ];

    public function setRisk(int $risk): void {
        if ($risk < 1 || $risk > 5) {
            throw new InvalidArgumentException("El nivel de riesgo debe estar entre 1 y 5.");
        }

        $this->risk = $risk;
        $this->setPercentage($this->riskPercentages[$risk]);
    }

    public function getRisk(): int {
        return $this->risk;
    }

    public function calculatedDeductions(): float {
        if ($this->getBaseSalary() <= 0) {
            throw new InvalidArgumentException("El salario base debe ser mayor a 0.");
        }

        if ($this->getPercentage() <= 0 || $this->getPercentage() > 1) {
            throw new InvalidArgumentException("El porcentaje debe estar entre 0 y 1.");
        }

        return $this->getBaseSalary() * $this->getPercentage();
    }
}

?>